tests/unit/client.py

```diff
@@ -0,0 +1,442 @@
+            x = ClientBase()
+        except LookupError:
+            assert True
+
+# FIX ME: The test may vary based on which backend session is used.
+#    def test_correct_init(self):
+#        x = ClientBase(ep, 'some_base64_string')
+#        assert x.endpoint == "http://127.0.0.1:8888"
+#        assert x.auth == "some_base64_string"
+
+    def test_object_url(self):
+        x = ClientBase(ep, 'some_base64_string')
+        y = x.object_url('abc', '123', 'xy23z')
+        assert y == 'http://127.0.0.1:8888/abc/123/xy23z'
+
+# For testing the client library we need a running HIL server, with dummy
+# objects populated. Following classes accomplish that end.
+# It shall:
+#       1. Configures haas.cfg
+#       2. Instantiates a database
+#       3. Starts a server on an arbitary port
+#       4. Populates haas with dummy objects
+#       5. tears down the setup in a clean fashion.
+
+
+# pytest.fixture(scope="module")
+
+def make_config():
+    """ This function creates haas.cfg with desired options
+    and writes to a temporary directory.
+    It returns a tuple where (tmpdir, cwd) = ('location of haas.cfg', 'pwdd')
+    """
+    tmpdir = tempfile.mkdtemp()
+    cwd = os.getcwd()
+    os.chdir(tmpdir)
+    with open('haas.cfg', 'w') as f:
+        config = '\n'.join([
+            '[general]'
+            'log_level = debug',
+
+            '[auth]',
+            'require_authentication = False',
+
+            '[headnode]',
+            'base_imgs = base-headnode, img1, img2, img3, img4',
+            '[database]',
+            'uri = sqlite:///%s/haas.db' % tmpdir,
+            '[extensions]',
+            'haas.ext.auth.null =',
+            'haas.ext.network_allocators.null =',
+            '[extensions]',
+            'haas.ext.switches.mock =',
+            'haas.ext.auth.null =',
+            'haas.ext.switches.nexus =',
+            'haas.ext.switches.dell =',
+            'haas.ext.switches.brocade =',
+            'haas.ext.obm.mock =',
+            'haas.ext.obm.ipmi =',
+            'haas.ext.network_allocators.null =',
+            '[haas.ext.network_allocators.vlan_pool]',
+            'vlans = 1001-1040',
+
+        ])
+        f.write(config)
+        return (tmpdir, cwd)
+
+
+def cleanup((tmpdir, cwd)):
+    """ Cleanup crew, when all tests are done.
+    It will shutdown the haas server,
+    delete any files and folders created for the tests.
+    """
+
+    os.remove('haas.cfg')
+    os.remove('haas.db')
+    os.chdir(cwd)
+    os.rmdir(tmpdir)
+
+
+def initialize_db():
+    """ Creates an  database as defined in haas.cfg."""
+    check_call(['haas-admin', 'db', 'create'])
+
+
+def run_server(cmd):
+    """This function starts a haas server.
+    The arguments in 'cmd' will be a list of arguments like required to start a
+    haas server like ['haas', 'serve', '8888']
+    It will return a handle which can be used to terminate the server when
+    tests finish.
+    """
+    proc = Popen(cmd)
+    return proc
```

Comment:
This function is literally just `Popen`.

The keystone ci tests arlready have some fixtures for running a real haas server; these should probably be moved to test_common and used here as well. We don't need to use the CLI to launch stuff here since it's a test of the client library, but we may want to elsewhere do some basic tests that the server responds when launched via the CLI. Unit tests should be as self-contained as possible though.